src/Makefile

```diff
@@ -719,8 +729,11 @@ endif
 ### needs access to the optimization flags.
 ifeq ($(optimize),yes)
 ifeq ($(debug), no)
-	ifeq ($(comp),clang)
+        ifeq ($(comp),$(filter $(comp),clang icx))
 		CXXFLAGS += -flto=full
+		ifeq ($(comp),icx)
+			CXXFLAGS += -fwhole-program-vtables
```

Comment:
I've been using icx with "-mtune=native" for optimization.  This should probably be looked into more to make use of icx.  
The current format of the Makefile I believe makes -march=... unnecessary given the explicit identification of instructions/extensions while still allowing to compile a binary to be compatible with various architectures.  The current format of the Makefile doesn't make it easy to specify the microarchitecture that the compiler should optimize for, and so I've simply been using native.